storage/innobase/xtrabackup/src/xtrabackup.cc

```diff
@@ -8622,6 +8622,24 @@ handle_options(int argc, char **argv, int *argc_client, char ***argv_client,
 		opt_password = get_tty_password(NullS);
 	}
 
+	msg("xtrabackup: Provided arguments:");
```

Comment:
Would be nice to add basic test case for this feature. Can you please also provide example output?
I have a feeling that this feature actually won't print used arguments, but rather it will print unused arguments... There is another feature -- `--print-param`, maybe we could use the same idea here? Also, do we need this for 2.3 as well?

Reply:
* The jira ticket only mentioned 2.4 - but I can open a PR for 2.3 too
* It prints out every argument, even the program name, and then every argument it detects as a client or server argument - for those, it even prints out the arguments it added automatically.
* I added a testcase

An example output:
```
./xtrabackup --backup --target-dir=/backups/ --some-arg
xtrabackup: Provided arguments: ./xtrabackup --backup --target-dir=/backups/ --some-arg
xtrabackup: Client arguments: --some-arg
xtrabackup: Server arguments: --user=mysql --pid-file=/var/run/mysqld/mysqld.pid --socket=/var/run/mysqld/mysqld.sock --port=3306 --basedir=/usr --lc-messages-dir=/usr/share/mysql --skip-external-locking --bind-address=127.0.0.1 --key_buffer_size=16M --max_allowed_packet=16M --thread_stack=192K --thread_cache_size=8 --myisam-recover-options=BACKUP --max_connections=10000 --query_cache_limit=1M --query_cache_size=16M --log_error=/var/log/mysql/error.log --expire_logs_days=10 --max_binlog_size=100M --backup --target-dir=/backups/ --some-arg
```